torch/csrc/distributed/rpc/init.cpp

```diff
@@ -356,32 +356,34 @@ If the future completes with an error, an exception is thrown.
   module.def(
       "_invoke_rpc_torchscript",
       [](const std::string& dstWorkerName,
-         const std::string& qualifiedNameStr,
-         const py::args& args,
-         const py::kwargs& kwargs) {
-        DCHECK(!PyGILState_Check());
+         const py::object& userCallable,
+         const py::tuple& argsTuple,
+         const py::dict& kwargsDict) {
+        DCHECK(PyGILState_Check());
         // No need to catch exception here, if function can not be found,
         // exception will be thrown in get_function() call; if args do not match
         // with function schema, exception will be thrown in
         // createStackForSchema() call.
+        auto qualifiedNameStr = c10::QualifiedName(
+            py::cast<std::string>(py::module::import("torch.jit")
+                                      .attr("_qualified_name")(userCallable)));
         auto qualifiedName = c10::QualifiedName(qualifiedNameStr);
         auto functionSchema = PythonRpcHandler::getInstance()
```

Comment:
this client thread will hold lock while server thread is trying to acquire the lock in request_callback, so we need to release the lock below

Reply:
@zhaojuanmao If client side get GIL, then Clients side should go and create the instance, right?

It looks to me a race, sine two threads can initialize 
```
static PythonRpcHandler* handler = new PythonRpcHandler();
```